File: insertion_sort.js
function insertionSort(array){
	for(var i = 1; i < array.length; i++){
		var temp = array[i];
		for(var j = i; j > 0 && temp < array[j-1]; j--){
			array[j] = array[j-1];
		}
		array[j] = temp;
	}
	return array;
}


var myArray = [5,4,3,2,1];
insertionSort(myArray);
console.log(myArray);

File: insertion_sort.rb
def insertion_sort(arr)
	for i in 1...arr.length
		temp = arr[i]
		j = i

		while j > 0 && temp < arr[j-1]
			arr[j] = arr[j-1]
			j -= 1
		end
		arr[j] = temp
	end
	arr
end

arr = Array.new(10) { rand(1..99) }

puts "before #{arr}"
insertion_sort(arr)
puts "after #{arr}"